frontend/dockerfile/dockerfile_test.go

```diff
@@ -6526,12 +6526,13 @@ COPY Dockerfile \
 
 func testReproSourceDateEpoch(t *testing.T, sb integration.Sandbox) {
 	integration.CheckFeatureCompat(t, sb, integration.FeatureOCIExporter, integration.FeatureSourceDateEpoch)
-	if sb.Snapshotter() == "native" {
-		t.Skip("the digest is not reproducible with the \"native\" snapshotter because hardlinks are processed in a different way: https://github.com/moby/buildkit/pull/3456#discussion_r1062650263")
+	if cdAddress := sb.ContainerdAddress(); cdAddress != "" {
+		// https://github.com/containerd/containerd/commit/9c9f564a35df7990870a53a125afbf88ac412753
+		integration.CheckContainerdVersion(t, cdAddress, ">= 1.7.0-beta.1")
```

Comment:
It looks good for containerd worker but I'm not sure what we should do for dockerd with containerd snapshotter enabled.

Reply:
Maybe having a factory in the integration sandbox to return the containerd version depending on the worker if that makes sense?

Reply:
I guess it can be revisited in a separate PR?
The current code is passing the current CI.